ga4gh/rnaseq2ga.py

```diff
@@ -246,78 +243,40 @@ def writeExpressionTable(writer, data):
+    rnaDB = RNASqliteStore(sqlFilename, dataset)
+    if rnaType == "cufflinks":
+        writer = CufflinksWriter(rnaDB, featureType)
+    elif rnaType == "kallisto":
+        writer = KallistoWriter(rnaDB, featureType)
+    elif rnaType == "rsem":
+        writer = RsemWriter(rnaDB, featureType)
+    else:
+        raise Exception("Unknown RNA file type: {}".format(rnaType))
+    readGroupSetName = readGroupSetNames.strip().split(",")[0]
+    featureSetIds = ""
+    readGroupIds = ""
+    if dataset:
+        featureSetIdList = []
+        for annotationId in annotationIds.split(","):
             featureSet = dataset.getFeatureSetByName(annotationId)
-            if rnaType == "cufflinks":
-                writer = CufflinksWriter(rnaDB, repoPath)
-            elif rnaType == "kallisto":
-                writer = KallistoWriter(rnaDB, repoPath)
-            elif rnaType == "rsem":
-                writer = RsemWriter(rnaDB, repoPath, featureType)
-            else:
-                raise Exception("Unknown RNA file type: {}".format(rnaType))
-            rnaQuantId = fields[0].strip()
-            quantFilename = os.path.join(dataFolder, fields[1].strip())
-            readGroupSetName = fields[4].strip()
-            readGroupSet = dataset.getReadGroupSetByName(readGroupSetName)
-            readGroupIds = ",".join(
-                [x.getId() for x in readGroupSet.getReadGroups()])
-            description = fields[5].strip()
-            programs = fields[6].strip()
-            writeRnaseqTable(rnaDB, [rnaQuantId], description,
-                             featureSet.getId(),
-                             readGroupId=readGroupIds, programs=programs)
-            quantFile = open(quantFilename, "r")
-            writeExpressionTable(writer, [(rnaQuantId, quantFile)])
-
-
-def parseArgs():
-    parser = argparse.ArgumentParser(
-        description="Script to generate SQLite database corresponding to "
-        "input RNA Quantification experiment files.")
-    parser.add_argument(
-        "--outputFile", "-o", default="rnaseq.db",
-        help="The file to output the server-ready database to.")
-    parser.add_argument(
-        "--inputDir", "-i",
-        help="Path to input directory containing RNA quant files.",
-        default='.')
-    parser.add_argument(
-        "--controlFile", "-c",
-        help="Name of control file (.tsv format) in the inputDir",
-        default="rna_control_file.tsv")
-    parser.add_argument('--verbose', '-v', action='count', default=0)
-    args = parser.parse_args()
-    return args
-
-
-@utils.Timed()
-def main():
-    args = parseArgs()
-    rnaseq2ga(args.inputDir, args.outputFile, args.controlFile)
-
-
-if __name__ == '__main__':
-    main()
+            featureSetIdList.append(featureSet.getId())
+        featureSetIds = ",".join(featureSetIdList)
+        # TODO: multiple readGroupSets
+        readGroupSet = dataset.getReadGroupSetByName(readGroupSetName)
```

Comment:
It would be good to gracefully fail if not found. I guess we get an uncaught exception?